zipline/finance/trading.py

```diff
@@ -135,55 +148,56 @@ def __init__(self, period_start, period_end,
         # copied to algorithm's environment for runtime access
         self.arena = arena
 
-        if trading_schedule is not None:
-            self.update_internal_from_trading_schedule(
-                trading_schedule=trading_schedule
-            )
-
-    def update_internal_from_trading_schedule(self, trading_schedule):
-
-        assert self.period_start <= self.period_end, \
-            "Period start falls after period end."
+        self.trading_calendar = trading_calendar
 
-        assert self.period_start <= trading_schedule.last_execution_day, \
-            "Period start falls after the last known trading day."
-        assert self.period_end >= trading_schedule.first_execution_day, \
-            "Period end falls before the first known trading day."
+        if self.period_start not in trading_calendar.schedule.index:
+            # if the start date is not a valid session in this calendar,
+            # push it forward to the first valid session
```

Comment:
Can you clarify?  You mean, have a parameter in the function call that specifies whether we push forward?